src/states/node.rs

```diff
@@ -1502,10 +1504,10 @@ impl Node {
             return Ok(());  // Avoid swarming back out to our own group.
         }
 
-        let target_peer_ids = self.get_targets(routing_msg, route, *hop)?;
+        let (new_sent_to, target_peer_ids) = self.get_targets(routing_msg, route, hop, sent_to)?;
```

Comment:
I wonder whether we can leave `sent_to` empty whenever we're not in the destination group?